server/consumer.go

```diff
@@ -1591,6 +1591,22 @@ func (o *consumer) updateConfig(cfg *ConsumerConfig) error {
 		}
 	}
 
+	if o.cfg.FilterSubject != cfg.FilterSubject {
+		if cfg.FilterSubject != _EMPTY_ && subjectHasWildcard(cfg.FilterSubject) {
+			o.filterWC = true
+		}
+		subject := cfg.FilterSubject
+		if subject == _EMPTY_ {
+			subject = fwcs
+		}
+		sub := &subscription{subject: []byte(subject), icb: o.processStreamSignal}
+		o.mset.clsMu.Lock()
+		o.mset.csl.Remove(o.sigSub)
+		o.mset.csl.Insert(sub)
+		o.sigSub = sub
+		o.mset.clsMu.Unlock()
```

Comment:
mset does some accounting here we need to track.

See func (mset *stream) removeConsumer(o *consumer)

So might want to make stream func

```go
func (mset *Stream) swapSigSubs(o *consumer)
```

Also, make sure o.sigSub != nil before calling remove I think.